tests/RobotFramework/tests/plugin_apt/install_overwrite_config_files.robot

```diff
@@ -0,0 +1,52 @@
+*** Settings ***
+Resource            ../../resources/common.resource
+Library             ThinEdgeIO
+Library    Cumulocity
+
+Suite Setup         Custom Setup
+Suite Teardown      Get Logs
+
+Test Tags           theme:software    theme:plugins
+
+
+*** Test Cases ***
+Install packages without overwriting config files
+    Execute Command    tedge config set apt.dpk.options.config keepold
+    # install package v1
+    Execute Command   /etc/tedge/sm-plugins/apt install /setup/sampledeb_1.0.0_all.deb  
+    # Modify the config file
+    Execute Command    echo "Updated the config" | sudo tee -a /etc/sampledeb.cfg
+
+    # install package v2
+    Execute Command   /etc/tedge/sm-plugins/apt install /setup/sampledeb_2.0.0_all.deb  
+    # Assert to make sure installation of newer version did not update the config file
+    ${output}=    Execute Command    cat /etc/sampledeb.cfg
+    Should Contain    ${output}    conf 1.0
+    Should Contain    ${output}    Updated the config
+    ThinEdgeIO.Remove Package Using APT    sampledeb
```

Comment:
Clean up commands should be run in a Teardown block because these steps won't be run if your previous assertion (e.g. `Should Contain`) fails, which might lead to unexpected issues in other tests since you are reusing the same device in both tests.

Reply:
Moved these steps to `tear down`

Reply:
Though I think you should use the Test Teardown and not the Suite Teardown as you want to make the device state is in the expected state before running the second test.

Reply:
right, I have fixed that